lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/file/download/v2/DownloadListDaoImplTest.java

```diff
@@ -322,17 +352,724 @@ public void testClearDownloadListWithMultipleUsers() {
+				new DownloadListItem().setFileEntityId(files.get(5).getId()).setVersionNumber(null));
+
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, userOneItems);
+
+		List<DownloadListItemResult> expected = downloadListDao.getDownloadListItems(userOneIdLong, userOneItems.get(3),
+				userOneItems.get(4));
+
+		List<Sort> sort = Arrays.asList(new Sort().setField(SortField.synId).setDirection(SortDirection.ASC));
+		Long limit = 2L;
+		Long offset = 3L;
+		// call under test
+		List<DownloadListItemResult> result = downloadListDao.getFilesAvailableToDownloadFromDownloadList(l -> l,
+				userOneIdLong, sort, limit, offset);
+		assertEquals(expected, result);
+	}
+	
+	@Test
+	public void testGetFilesAvailableToDownloadFromDownloadListWithOrderByProjectName() {
+		int numberOfProject = 2;
+		int foldersPerProject = 1;
+		int filesPerFolder = 3;
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+		assertEquals(6, files.size());
+
+		// Add the latest version of each file to user's download list
+		List<DownloadListItem> userOneItems = files.stream()
+				.map(f -> new DownloadListItem().setFileEntityId(f.getId()).setVersionNumber(null))
+				.collect(Collectors.toList());
+
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, userOneItems);
+
+		List<DownloadListItemResult> expected = downloadListDao.getDownloadListItems(userOneIdLong,
+				userOneItems.get(3),
+				userOneItems.get(4),
+				userOneItems.get(5),
+				userOneItems.get(0),
+				userOneItems.get(1),
+				userOneItems.get(2)
+		);
+
+		List<Sort> sort = Arrays.asList(
+				new Sort().setField(SortField.projectName).setDirection(SortDirection.DESC),
+				new Sort().setField(SortField.synId).setDirection(SortDirection.ASC));
+		Long limit = 100L;
+		Long offset = 0L;
+		// call under test
+		List<DownloadListItemResult> result = downloadListDao.getFilesAvailableToDownloadFromDownloadList(l -> l,
+				userOneIdLong, sort, limit, offset);
+		assertEquals(expected, result);
+	}
+	
+	@Test
+	public void testGetFilesAvailableToDownloadFromDownloadListWithMultipleVersionOfSameFile() {
+		int numberOfProject = 1;
+		int foldersPerProject = 1;
+		int filesPerFolder = 1;
+		List<Node> files = createFileHierarchy(numberOfProject, foldersPerProject, filesPerFolder);
+		assertEquals(1, files.size());
+
+		List<DownloadListItem> userOneItems = Arrays.asList(
+				new DownloadListItem().setFileEntityId(files.get(0).getId()).setVersionNumber(1L),
+				new DownloadListItem().setFileEntityId(files.get(0).getId()).setVersionNumber(2l),
+				new DownloadListItem().setFileEntityId(files.get(0).getId()).setVersionNumber(null)
+		);
+
+		downloadListDao.addBatchOfFilesToDownloadList(userOneIdLong, userOneItems);
+
+		List<DownloadListItemResult> expected = downloadListDao.getDownloadListItems(userOneIdLong,
+				userOneItems.get(2),
+				userOneItems.get(0),
+				userOneItems.get(1)
+		);
+
+		List<Sort> sort = null;
```

Comment:
We might want to add a default sort if it is not specified, e.g. by entity id, version ASC. Additionally we could expose in the API the version sort field.